### PR TITLE
Add cluster_kms_key_deletion_window_in_days parameter

### PR DESCRIPTION
**Pull request by ruzickap**
### What does this PR do?

Adds possibility to specify the period for deleting the KMS key.

### Motivation

By default - the KMS key created by the module will remain in AWS for 30 days. For some dev/test clusters which are being created multiple times per dat this value is too high.

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/ssp-eks-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by vara-bonthu — APPROVED** [review]
Thanks for you contribution @ruzickap !!  PR Looks good to me
I will wait for another approval before merging your PR

**Review by askulkarni2 — APPROVED** [review]
LGTM!

File: variables.tf
Comment (ruzickap) [reply variables.tf:98]: I took the variable name from [aws-kms](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/modules/aws-kms) module which is part of this repository: https://github.com/aws-samples/aws-eks-accelerator-for-terraform/blob/002ef82d8114f03d79880a37f8409a4d93ddb150/modules/aws-kms/variables.tf#L40

The variable names should be very similar in the "sub module" [aws-kms](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/modules/aws-kms) and main terraform code to allow better readability/maintainability.

Anyway - if you really want to change it let me know... (in such case it should be considered to change it also in the [aws-kms](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/modules/aws-kms) module)

**Pull request merged**
